docs/source/app-configuration.rst

```diff
@@ -0,0 +1,160 @@
+App Integration Configuration
+=============================
+
+Overview
+--------
+
+To further StreamAlert's abilities past ingesting logs from services that offer support for sending to AWS Lambda,
+the package includes a method to fetch logs from RESTful APIs to be forwarded to StreamAlert. This capability is added through the
+use of StreamAlert Apps.
+
+StreamAlert Apps are designed to run on scheduled intervals and collect logs from a third-party API to be relayed on to the Rule Processor
+for processing.
+
+
+Concepts
+--------
+
+StreamAlert Apps are made possible through the use of AWS technologies:
+
+* `AWS EC2 Simple System Manager's Parameter Store <https://aws.amazon.com/ec2/systems-manager/parameter-store/>`_
+* `AWS Lambda Invocations via Scheduled Events <http://docs.aws.amazon.com/lambda/latest/dg/with-scheduled-events.html>`_
+
+
+Supported Services
+------------------
+
+* Duo
+
+  - Authentication Logs
+  - Administrator Logs
+
+
+Getting Started
+---------------
+
+An initial deploy of StreamAlert must be performed before StreamAlert Apps can be configured. If you do not have a current deploy,
+please visit the `Project's Getting Started <getting-started.html>`_ page to get up and running.
+
+
+Adding a new StreamAlert App requires the following steps:
+
+1. Configure the StreamAlert App through the CLI (via ``python manage.py app new``).
+2. Enter the required authentication information for the app being configured.
+3. Deploy the new App and the Rule Processor to accept incoming data from your App.
+
+To get help configuring a new StreamAlert App, use:
+
+.. code-block:: bash
+
+  $ python manage.py app new --help
+
+
+1. Configure the StreamAlert App
+````````````````````````````````
+
+The StreamAlert CLI is used to add a new App configuration.
+
+.. code-block:: bash
+
+  $ python manage.py app new \
```

Comment:
i presume "safe" defaults are already set and therefore the end user doesn't need to paste this entire thing?

Reply:
No - all of these flags are required and there are no defaults (by design). To explain, we can't readily recommend a default that is suitable for everyone's scenario since this depends on multiple variables:

1) How many logs that xxxx service is creating
2) How quickly they care to know about those logs (ie: running every 10 minutes or every 10 days)
3) If the interval is very low (10 minutes) then the amount of logs collected should be low - thus timeout can probably also be low.
4) If the interval is very high (10 days) then the amount of logs collected will be significantly larger - thus timeout should be much higher.

When you extrapolate these factors across all of the potential services we will support, this will be very hairy to make 'safe'. Each of them may rate limit their API at different max request values, each may return a different max amount of logs in each request, each may require transforming the incoming data in different ways before being shoveled to the Rule Processor, etc.

I'm open to discussing further if you'd like, but I was not prepared to account for all of the above for v0.0.1.

Reply:
Sorry if I made this confusing, I was just thinking about default settings on a per-app basis. But, as you point out, even if we did that for DUO, very very large deployments will have a lot more logs.

Reply:
In a perfect world, we could have a way (like we discussed) to help users analyze their situation via some form of log sampling for each service. But we're still pretty far off from that. I hesitate to make any suggestions and cause issues when others (hopefully) know their situation better and can adjust accordingly.